### PR TITLE
Iht 4322 Content changes for Gifts, Exemptions, Debts and Tnrb

### PR DESCRIPTION
**Pull request by tyagikumarvineet**

### CODE REVIEW

File: app/iht/utils/CommonHelper.scala [hunk 1]
```diff
@@ -454,4 +454,7 @@ object CommonHelper {
     }
   }
 
+  def getDeceasedNameOrDefaultString(regDetails: RegistrationDetails):String =
+      regDetails.deceasedDetails.fold(Messages("iht.the.deceased"))(deceased => deceased.name)
```
Comment (grantrobson) [inline app/iht/utils/CommonHelper.scala]: More concise to say _.name

File: app/iht/utils/tnrb/TnrbHelper.scala [hunk 2]
```diff
@@ -57,6 +57,19 @@ object TnrbHelper {
     }
   }
 
+  /**
+    * Returns the Spouse name (if exists) otherwise returns the pretext string
+    */
+  def spouseOrCivilPartnerName(tnrbModel: TnrbEligibiltyModel,
+                                prefixText: String=""): String  = {
+    if(tnrbModel.Name.toString.trim!=""){
```
Comment (grantrobson) [inline app/iht/utils/tnrb/TnrbHelper.scala]: More efficient and readable version:
    CommonHelper.withValue(tnrbModel.Name.toString.trim) {
      case name if name.isEmpty => prefixText
      case name => name
    }

File: app/iht/views/application/gift/given_away.scala.html [hunk 3]
```diff
@@ -36,14 +36,17 @@
 
     @helpers.form(action = iht.controllers.application.gifts.routes.GivenAwayController.onSubmit,'autoComplete -> "off") {
 
-    <p>@Messages("page.iht.application.gifts.lastYears.givenAway.p1",getDateBeforeSevenYears(getOrException(regDetails.deceasedDateOfDeath).dateOfDeath).toString(dateFormatForDisplay),getOrException(regDetails.deceasedDateOfDeath).dateOfDeath.toString(dateFormatForDisplay))</p>
-    <p>@Messages("page.iht.application.gifts.lastYears.givenAway.p2")</p>
+    <p>@Messages("page.iht.application.gifts.lastYears.givenAway.p1",
+                        getDeceasedNameOrDefaultString(regDetails),
```
Comment (grantrobson) [inline app/iht/views/application/gift/given_away.scala.html]: Maybe prefix getDeceasedNameOrDefaultString and getDateBeforeSevenYears to make it clear where these methods are defined. You have done elsewhere with the first.

File: test/iht/views/application/gifts/GivenAwayViewTest.scala [hunk 4]
```diff
@@ -51,16 +52,25 @@ class GivenAwayViewTest extends UnitSpec with FakeIhtApp with MockitoSugar with
...
       saveAndContinueLink.text shouldBe Messages("iht.saveAndContinue")
     }
 
-    "show the correct question" in {
+    "show the correct question and guidance" in {
       implicit val request = createFakeRequest()
-      val view = given_away(giftsGivenAwayForm, CommonBuilder.buildRegistrationDetailsWithDeceasedDetails)
-      view.toString should include(Messages("page.iht.application.gifts.lastYears.givenAway.question"))
+      val view = given_away(giftsGivenAwayForm, regDetails)
+      view.toString should include(Messages("page.iht.application.gifts.lastYears.givenAway.question",
```
Comment (grantrobson) [inline test/iht/views/application/gifts/GivenAwayViewTest.scala]: Store view.toString in a variable to save memory useage.

File: app/iht/controllers/application/assets/insurancePolicy/InsurancePolicyOverviewController.scala [hunk 5]
```diff
@@ -115,19 +116,20 @@ trait InsurancePolicyOverviewController extends EstateController {
...
-            "iht.estateReport.insurancePolicies.overLimitNotOwnEstate.question",
-            "iht.estateReport.assets.insurancePolicies.buyAnnuity.question",
-            "page.iht.application.assets.insurance.policies.overview.other.question4"
+            Messages("iht.estateReport.insurancePolicies.premiumsNotPayingOut.question"),
+            Messages("iht.estateReport.insurancePolicies.overLimitNotOwnEstate.question"),
+            Messages("iht.estateReport.assets.insurancePolicies.buyAnnuity.question"),
+            Messages("page.iht.application.assets.insurance.policies.overview.other.question4")
           ),
           ad, regDetails)
 //          //s"$questionTitlesMessagesFilePrefix.question$index"
```
Comment (tyagikumarvineet) [inline app/iht/controllers/application/assets/insurancePolicy/InsurancePolicyOverviewController.scala]: Done

File: app/iht/utils/CommonHelper.scala (hunk 1 above)
Comment (tyagikumarvineet) [inline app/iht/utils/CommonHelper.scala]: Done

File: app/iht/utils/tnrb/TnrbHelper.scala (hunk 2 above)
Comment (tyagikumarvineet) [inline app/iht/utils/tnrb/TnrbHelper.scala]: Done

File: app/iht/views/application/gift/given_away.scala.html (hunk 3 above)
Comment (tyagikumarvineet) [inline app/iht/views/application/gift/given_away.scala.html]: Done

File: test/iht/views/application/gifts/GivenAwayViewTest.scala (hunk 4 above)
Comment (tyagikumarvineet) [inline test/iht/views/application/gifts/GivenAwayViewTest.scala]: Done

File: app/iht/controllers/application/assets/insurancePolicy/InsurancePolicyOverviewController.scala [hunk 6]
```diff
@@ -102,8 +103,8 @@ trait InsurancePolicyOverviewController extends EstateController {
           questionLevelLinkAccessibilityTextValue = "page.iht.application.assets.insurance.policies.overview.deceased.amount.screenReader.link.value",
           questionAnswerExprYesNo = insurancePolicy.policyInDeceasedName,
           questionAnswerExprValue = insurancePolicy.value,
-          questionTitleYesNoMessageKey = "iht.estateReport.insurancePolicies.ownName.question",
-          questionTitleValueMessageKey = "iht.estateReport.assets.insurancePolicies.totalValueOwnedAndPayingOut"
+          questionTitleYesNoMessageKey = Messages("iht.estateReport.insurancePolicies.ownName.question"),
```
Comment (tyagikumarvineet) [inline app/iht/controllers/application/assets/insurancePolicy/InsurancePolicyOverviewController.scala]: Done

**Pull request merged**